### ISSUE TITLE
Many 500 errors

### ISSUE THREAD

**Issue by nottrobin** [canonical-web-and-design/www.ubuntu.com#2600]
TL;Dr: prevent URLs with special characters from causing 500 server errors - they should 404 instead.

Google sent an email about the shiny new interface for the Search Console. So I went to check it out. It said there are about 182 URLs that "error" - more than the number of successful URLs:

https://search.google.com/u/0/search-console/index/drilldown?resource_id=http%3A%2F%2Fwww.ubuntu.com%2F&item_key=CAMYEyAE

These are silly erroneous URLs like http://www.ubuntu.com/download%EC%97%90 - each of the returns a 500 server error.

I don't know where Google is getting these URLs, but it doesn't matter that much, we don't really need to worry since they're not important URLs. But they shouldn't really error, they should return 404 instead - then they wouldn't pollute our search console.

**Comment by nottrobin** [comment]
I looked into this. The problem is a character encoding one in the URLs (unsurprisingly). It's because in the context where the application is running on live, the default language is ASCII.

It can be fixed quite easily by setting the `LANG=c.UTF-8` when running Gunicorn - e.g. by defining it in the `/etc/init/gunicorn.conf` file. However, the [Gunicorn charm](https://jujucharms.com/gunicorn/) doesn't offer any way to configure this, so to achieve this in production we'd have to fork the Gunicorn charm just for this purpose.

As this issues isn't *actually* impacting users in any real way - it's only impacting search bots - I think we should hold off on fixing this until we move to hosting the site on Kubernetes, at which point this will be trivially fixed in the Dockerfile, as we [have on other projects](https://github.com/canonical-websites/snapcraft.io/blob/master/Dockerfile#L7).

**Comment by anasereijo** [comment]
@nottrobin is this still blocked?

**Comment by nottrobin** [comment]
We can close this. This will be fixed by moving to K8s.
